frontend/src/components/custom-tools/prompt-card/PromptCardItems.jsx

```diff
@@ -60,20 +61,24 @@ function PromptCardItems({
     isSimplePromptStudio,
     isPublicSource,
     adapters,
+    defaultLlmProfile,
   } = useCustomToolStore();
+  const { promptOutputs: promptOutput } = usePromptOutputStore();
```

Comment:
@jagadeeswaran-zipstack Can you check if using `promptOutputs` directly from the store is necessary. If `promptOutputs` are already being passed as props, accessing them from the store might cause the entire list of prompt cards to re-render whenever the store changes, negating the intended optimization of selective re-renders.

Check how the promptOutputs is handled in the [DocumentParser.jsx](https://github.com/Zipstack/unstract/blob/e6c025519c2c956e5c002e3c979254b920018ba1/frontend/src/components/custom-tools/document-parser/DocumentParser.jsx#L277) component.

Reply:
Understood. I'll make the required changes.